app/src/main/java/ch/epfl/sdp/Map/PathsHandler.java

```diff
@@ -0,0 +1,136 @@
+        CameraPosition position = new CameraPosition.Builder()
+                .target(new LatLng(latitude, longitude))
+                .zoom(7)
+                .build();
+        if (map != null) {
+            map.setCameraPosition(position);
+        }
+    }
+
+    private void getPathCoordinates(@NonNull Iterator<QueryDocumentSnapshot> qsIterator) {
+        // TODO: get path for given day, NEED TO RETRIEVE POSITIONS ON SPECIFIC DAY TIME
+        pathCoordinates = new ArrayList<>();
+
+        for (; qsIterator.hasNext(); ) {
+            QueryDocumentSnapshot qs = qsIterator.next();
+            try {
+                GeoPoint geoPoint = (GeoPoint)((Map)qs.get("Position")).get("geoPoint");
+                double lat = geoPoint.getLatitude();
+                double lon = geoPoint.getLongitude();
+                pathCoordinates.add(Point.fromLngLat(lon, lat));
+            } catch (NullPointerException ignored) {
+                Log.d("ERROR ADDING POINT", String.valueOf(ignored));
+            }
+        }
+
+        Log.d("PATH COORD LENGTH: ", String.valueOf(pathCoordinates.size()));
+        Log.d("IS PATH COORD NULL? ", (pathCoordinates == null) ? "YES" : "NO");
+        latitude = pathCoordinates.get(0).latitude();
+        longitude = pathCoordinates.get(0).longitude();
+        setPathLayer();
+    }
+
+    private void setPathLayer() {
+        Layer layer = new LineLayer(PATH_LAYER_ID, PATH_SOURCE_ID).withProperties(
+                PropertyFactory.lineCap(Property.LINE_CAP_ROUND),
+                PropertyFactory.lineWidth(5f),
+                PropertyFactory.lineColor(Color.parseColor("maroon"))
+        );
+        map.getStyle(style -> {
+
+            // Create the LineString from the list of coordinates and then make a GeoJSON
+            // FeatureCollection so we can add the line to our map as a layer.
+            style.addSource(new GeoJsonSource(PATH_SOURCE_ID,
+                    FeatureCollection.fromFeatures(new Feature[]{Feature.fromGeometry(
+                            LineString.fromLngLats(pathCoordinates)
+                    )})));
+
+            style.addLayer(layer);
+        });
+        layer.setProperties(visibility(NONE));
+    }
+
+    private void initFirestorePathRetrieval(Callback<Iterator<QueryDocumentSnapshot>> callback) {
+        db.collection("History/USER_PATH_DEMO/Positions")
+                //.orderBy("timestamp")
+                //.limit(50)
+                .get()
+                .addOnCompleteListener(task -> {
+                    if (task.isSuccessful()) {
+                        callback.onCallback(task.getResult().iterator());
+                    } else {
+                        Toast.makeText(parentClass.getActivity(),
+                                R.string.cannot_retrieve_positions,
+                                Toast.LENGTH_LONG).show();
+                    }
+                });
+    }
+
+    private void getInfectedMet() { // This function is not done yet
```

Comment:
Is it done now ?

Reply:
No, need to merge this branch first, to be able to add this feature on another branch.

Reply:
k